Tomes/TomeValue.cs

```diff
@@ -37,39 +37,66 @@ public class TomeValue : JValue {
 			TomeArray newTomeArray = new TomeArray((JArray)newValue, root);
 			this.Replace(newTomeArray);
 
-			onChanged -= EmitToParents;
-			onChanged += newTomeArray.onChanged;
-			newTomeArray.onChanged = onChanged;
-			newTomeArray.onDestroy = onDestroy;
-
-			if (newTomeArray.onChanged != null) {
-				newTomeArray.onChanged.Invoke(null);
+			if (this.Parent == null) {
+				// If replace was successfuly move over event handlers and call new onChanged handler
+				// The instance in which replace would not be successful, is when the old and new values are the same
+				onChanged -= EmitToParents;
+				onChanged += newTomeArray.onChanged;
+				newTomeArray.onChanged = onChanged;
+				newTomeArray.onDestroy = onDestroy;
+
+				if (newTomeArray.onChanged != null) {
+					newTomeArray.onChanged.Invoke(null);
+				}
+			} else {
+				// Otherwise call original onChanged handler
+				if (onChanged != null) {
+					onChanged.Invoke(null);
+				}
 			}
 			break;
 		case JTokenType.Object:
 			TomeObject newTomeObject = new TomeObject((JObject)newValue, root);
 			this.Replace(newTomeObject);
 
-			onChanged -= EmitToParents;
-			onChanged += newTomeObject.onChanged;
-			newTomeObject.onChanged = onChanged;
-			newTomeObject.onDestroy = onDestroy;
-
-			if (newTomeObject.onChanged != null) {
-				newTomeObject.onChanged.Invoke(null);
+			if (this.Parent == null) {
+				// If replace was successfuly move over event handlers and call new onChanged handler
+				// The instance in which replace would not be successful, is when the old and new values are the same
+				onChanged -= EmitToParents;
+				onChanged += newTomeObject.onChanged;
+				newTomeObject.onChanged = onChanged;
+				newTomeObject.onDestroy = onDestroy;
+				
+				if (newTomeObject.onChanged != null) {
+					newTomeObject.onChanged.Invoke(null);
+				}
+			} else {
+				// Otherwise call original onChanged handler
+				if (onChanged != null) {
+					onChanged.Invoke(null);
+				}
 			}
 			break;
 		default:
 			TomeValue newTomeValue = new TomeValue((JValue)newValue, root);
 			this.Replace(newTomeValue);
 
-			onChanged -= EmitToParents;
-			onChanged += newTomeValue.onChanged;
-			newTomeValue.onChanged = onChanged;
-			newTomeValue.onDestroy = onDestroy;
-
-			if (newTomeValue.onChanged != null) {
-				newTomeValue.onChanged.Invoke(null);
+			if (this.Parent == null) {
+				// If replace was successfuly move over event handlers and call new onChanged handler
+				// The instance in which replace would not be successful, is when the old and new values are the same
+				onChanged -= EmitToParents;
+				onChanged += newTomeValue.onChanged;
+				newTomeValue.onChanged = onChanged;
+				newTomeValue.onDestroy = onDestroy;
+				
+				if (newTomeValue.onChanged != null) {
+					newTomeValue.onChanged.Invoke(null);
+				}
+			} else {
+				// Otherwise call original onChanged handler
+				if (onChanged != null) {
+					onChanged.Invoke(null);
+				}
```

Comment:
since we not longer depend on a cast here so should be possible
touching this spot tends to break a lot quite easily but hard to debug
mind if we make an issue for this one and i will do it later